doc/build/dependency.md

```diff
@@ -0,0 +1,94 @@
+# Building dependencies
```

Comment:
I think we can move (and update) the file ./build/android/README.md into this documentation folder or merge it with this file. We should update it because some lines are not more valid with the latest NDK. For example, with the suggested NDK_DIR the script doesn't find the build toolchain python script.